### ISSUE TITLE
Exception when starting a new game

### ISSUE THREAD

**Issue by Gualdimar** [yairm210/Unciv#8589]
**Version**
The latest master branch

**Describe the bug**
Every now and then an error pops up when starting starting a new game on default settings an error pops up: 
"It looks like we can't make a map with the parameters you requested!
Maybe you put too many players into too small a map?"

<details><summary>Exception message</summary>

```
java.lang.IndexOutOfBoundsException: Index 100019 out of bounds for length 41
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	at java.base/java.util.Objects.checkIndex(Objects.java:372)
	at java.base/java.util.ArrayList.get(ArrayList.java:459)
	at com.unciv.logic.map.TileMap.get(TileMap.kt:153)
	at com.unciv.logic.map.TileMap.get(TileMap.kt:147)
	at com.unciv.logic.map.mapgenerator.MapRegions.findStart(MapRegions.kt:505)
	at com.unciv.logic.map.mapgenerator.MapRegions.assignRegions(MapRegions.kt:254)
	at com.unciv.logic.map.mapgenerator.MapGenerator$generateMap$11.invoke(MapGenerator.kt:146)
	at com.unciv.logic.map.mapgenerator.MapGenerator$generateMap$11.invoke(MapGenerator.kt:145)
	at com.unciv.logic.map.mapgenerator.MapGenerator.runAndMeasure(MapGenerator.kt:199)
	at com.unciv.logic.map.mapgenerator.MapGenerator.generateMap(MapGenerator.kt:145)
	at com.unciv.logic.GameStarter$startNewGame$3.invoke(GameStarter.kt:66)
	at com.unciv.logic.GameStarter$startNewGame$3.invoke(GameStarter.kt:63)
	at com.unciv.logic.GameStarter.runAndMeasure(GameStarter.kt:144)
	at com.unciv.logic.GameStarter.startNewGame(GameStarter.kt:63)
	at com.unciv.ui.newgamescreen.NewGameScreen$startNewGame$2.invokeSuspend(NewGameScreen.kt:268)
	at com.unciv.ui.newgamescreen.NewGameScreen$startNewGame$2.invoke(NewGameScreen.kt)
	at com.unciv.ui.newgamescreen.NewGameScreen$startNewGame$2.invoke(NewGameScreen.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
	at com.unciv.ui.newgamescreen.NewGameScreen.startNewGame(NewGameScreen.kt:259)
	at com.unciv.ui.newgamescreen.NewGameScreen.access$startNewGame(NewGameScreen.kt:46)
	at com.unciv.ui.newgamescreen.NewGameScreen$4$5.invokeSuspend(NewGameScreen.kt:196)
	at com.unciv.ui.newgamescreen.NewGameScreen$4$5.invoke(NewGameScreen.kt)
	at com.unciv.ui.newgamescreen.NewGameScreen$4$5.invoke(NewGameScreen.kt)
	at com.unciv.utils.concurrency.ConcurrencyKt$launchCrashHandling$1.invokeSuspend(Concurrency.kt:83)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:164)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:164)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandling$1.invoke(CrashHandlingExtensions.kt:17)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher.dispatch$lambda$0(Concurrency.kt:164)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```
</details>

**Comment by Gualdimar** [comment]
<details><summary>Debug log</summary>

```
2023-02-05T17:18:23.816696700Z [main] [UncivGame] Setting new screen: com.unciv.MainMenuScreen@489110c2, screenStack: [com.unciv.ui.worldscreen.WorldScreen@272778ae, com.unciv.MainMenuScreen@489110c2]
2023-02-05T17:18:24.227511Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerator] Mountain-like generation for Mountain
2023-02-05T17:18:24.227511Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerator] Hill-like generation for Mountain
2023-02-05T17:18:25.043932700Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 1 < initial 8
2023-02-05T17:18:25.136947800Z [threadpool-daemon-2 @ShowMapBackground#49] [NaturalWonderGenerator] Natural Wonder Cerro de Potosi @(1.0,3.0)
2023-02-05T17:18:25.137952300Z [threadpool-daemon-2 @ShowMapBackground#49] [NaturalWonderGenerator] Natural Wonder Old Faithful @(-6.0,-1.0)
2023-02-05T17:18:25.137952300Z [threadpool-daemon-2 @ShowMapBackground#49] [NaturalWonderGenerator] Natural Wonders for this game: [Cerro de Potosi, Old Faithful]
2023-02-05T17:18:25.138948300Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 8 < initial 11
2023-02-05T17:18:25.139948400Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 8 < initial 12
2023-02-05T17:18:25.139948400Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 9 < initial 12
2023-02-05T17:18:25.140948200Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 9 < initial 11
2023-02-05T17:18:25.140948200Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 9 < initial 11
2023-02-05T17:18:25.144948600Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 2 < initial 4
2023-02-05T17:18:25.146949200Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 2 < initial 4
2023-02-05T17:18:25.147948400Z [threadpool-daemon-2 @ShowMapBackground#49] [MapGenerationRandomness] chooseSpreadOutLocations: distance 7 < initial 11
2023-02-05T17:18:25.500205500Z [main] [UncivGame] Setting new screen: com.unciv.ui.newgamescreen.NewGameScreen@13f4048e, screenStack: [com.unciv.ui.worldscreen.WorldScreen@272778ae, com.unciv.MainMenuScreen@489110c2, com.unciv.ui.newgamescreen.NewGameScreen@13f4048e]
2023-02-05T17:18:27.144601100Z [threadpool-nondaemon-0 @NewGame#53] [MapGenerator] Mountain-like generation for Mountain
2023-02-05T17:18:27.144601100Z [threadpool-nondaemon-0 @NewGame#53] [MapGenerator] Hill-like generation for Mountain
2023-02-05T17:18:28.743264700Z [threadpool-nondaemon-0 @NewGame#53] [MapGenerationRandomness] chooseSpreadOutLocations: distance 2 < initial 10
java.lang.IndexOutOfBoundsException: Index 100019 out of bounds for length 41
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	at java.base/java.util.Objects.checkIndex(Objects.java:372)
	at java.base/java.util.ArrayList.get(ArrayList.java:459)
	at com.unciv.logic.map.TileMap.get(TileMap.kt:153)
	at com.unciv.logic.map.TileMap.get(TileMap.kt:147)
	at com.unciv.logic.map.mapgenerator.MapRegions.findStart(MapRegions.kt:505)
	at com.unciv.logic.map.mapgenerator.MapRegions.assignRegions(MapRegions.kt:254)
	at com.unciv.logic.map.mapgenerator.MapGenerator$generateMap$11.invoke(MapGenerator.kt:146)
	at com.unciv.logic.map.mapgenerator.MapGenerator$generateMap$11.invoke(MapGenerator.kt:145)
	at com.unciv.logic.map.mapgenerator.MapGenerator.runAndMeasure(MapGenerator.kt:199)
	at com.unciv.logic.map.mapgenerator.MapGenerator.generateMap(MapGenerator.kt:145)
	at com.unciv.logic.GameStarter$startNewGame$3.invoke(GameStarter.kt:66)
	at com.unciv.logic.GameStarter$startNewGame$3.invoke(GameStarter.kt:63)
	at com.unciv.logic.GameStarter.runAndMeasure(GameStarter.kt:144)
	at com.unciv.logic.GameStarter.startNewGame(GameStarter.kt:63)
	at com.unciv.ui.newgamescreen.NewGameScreen$startNewGame$2.invokeSuspend(NewGameScreen.kt:268)
	at com.unciv.ui.newgamescreen.NewGameScreen$startNewGame$2.invoke(NewGameScreen.kt)
	at com.unciv.ui.newgamescreen.NewGameScreen$startNewGame$2.invoke(NewGameScreen.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
	at com.unciv.ui.newgamescreen.NewGameScreen.startNewGame(NewGameScreen.kt:259)
	at com.unciv.ui.newgamescreen.NewGameScreen.access$startNewGame(NewGameScreen.kt:46)
	at com.unciv.ui.newgamescreen.NewGameScreen$4$5.invokeSuspend(NewGameScreen.kt:196)
	at com.unciv.ui.newgamescreen.NewGameScreen$4$5.invoke(NewGameScreen.kt)
	at com.unciv.ui.newgamescreen.NewGameScreen$4$5.invoke(NewGameScreen.kt)
	at com.unciv.utils.concurrency.ConcurrencyKt$launchCrashHandling$1.invokeSuspend(Concurrency.kt:83)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:164)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:164)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandling$1.invoke(CrashHandlingExtensions.kt:17)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher.dispatch$lambda$0(Concurrency.kt:164)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```
</details>

**Comment by vegeta1k95** [comment]
@yairm210 
This maybe related to HexMath changes.

**Comment by yairm210** [comment]
Possibly?

**Comment by yairm210** [comment]
What settings are you running this on? I tried loads of tries and never got this

**Comment by Gualdimar** [comment]
Defaults:
```
lastGameSetup:{gameParameters:{players:[{playerType:Human},{},{},{}],victoryTypes:[Scientific,Cultural,Domination,Diplomatic,Time]},mapParameters:{mapSize:{radius:20,width:44,height:29,name:Medium},createdWithVersion:4.4.10 (Build 807),seed:1675628210342}},version:1}
```

**Comment by yairm210** [comment]
Okay, so map size medium, what shape? what type?

**Comment by Gualdimar** [comment]
oh, sorry, I thought if you change the settings, they are written in GameSettings.json.
so, 
map type: generated
map shape: hexagonal
map generation type: pangaea(just tried Flat earth and got error on first try)
world size: medium
resource setting: default
all cheboxes - unchecked

**Comment by Gualdimar** [comment]
rectangular, three continents - first try

**Comment by Gualdimar** [comment]
i can't start new game at all with these types of map:
Hexagonal
    Continent and islands
    Two continents
    Three continents
    
Flat earth
    Continent and islands
    Two continents
    Three continents
    Four corners
    
Rectangular
    Continent and islands
    Two continents
    Three continents
    four corners

**Comment by yairm210** [comment]
Solved in f6741229
